File: leetcode-submissions/1267_Count_Servers_that_Communicate.ts
function countServers(grid: number[][]): number {
let resultantServers:number =0;
let m:number =grid.length;
let n:number =grid[0].length;
for(let i:number=0;i<m;i++)
{
    for(let j:number =0;j<n;j++){
        if(grid[i][j]==1){
            let isAnotherServer:boolean =false;
            //check in the row 
            for(let row:number =0;row<m;row++){
                if(row!==i && grid[row][j]===1){
                    isAnotherServer=true;
                    break;
                }
            }
            if(isAnotherServer){
                resultantServers++;
                continue;
            }
            for(let col:number =0;col<n;col++){
                if(col!==j && grid[i][col]===1){
                    isAnotherServer=true;
                    break;
                }
            }
            if(isAnotherServer){
                resultantServers++;
            }
        }
    }
}
return resultantServers++;

};

File: leetcode-submissions/1267_Count_Servers_that_Communicate.cpp
class Solution {
public:
    int countServers(vector<vector<int>>& grid) {
        int m=grid.size();
        int n=grid[0].size();

        vector<int>IndexColCount(n,0);
        vector<int>IndexRowCount(m,0);
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(grid[i][j]==1){
                    IndexColCount[j]++;
                    IndexRowCount[i]++;
                }
            }
        }
        int resultantServers=0;

        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                if(grid[i][j]==1 and (IndexRowCount[i]>1 or IndexColCount[j]>1)){
                    resultantServers++;
                }
            }
        }
        return resultantServers;
    }
};